demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -474,6 +486,77 @@ def run_validations_on_file(self, file_path, pack_error_ignore_list, is_modified
 
         return True
 
+    @staticmethod
+    def get_file_by_status(modified_files: Set, old_format_files: Set,
+                           file_path: str) -> Tuple[Set, Set, Set]:
+        """Given a specific file path identify in which git status set
+        it exists and return a set containing that file and 2 additional empty sets."""
+        filtered_modified_files: Set = set()
+        filtered_added_files: Set = set()
+        filtered_old_format: Set = set()
+
+        # go through modified files and try to identify if the file is there
+        for file in modified_files:
+            if isinstance(file, str) and file == file_path:
+                filtered_modified_files.add(file_path)
+                return filtered_modified_files, filtered_added_files, filtered_old_format
+
+            # handle renamed files which are in tuples
+            elif file_path in file:
+                filtered_modified_files.add(file)
+                return filtered_modified_files, filtered_added_files, filtered_old_format
+
+        # if the file is not modified check if it is in old format files
+        if file_path in old_format_files:
+            filtered_old_format.add(file_path)
+
+        else:
+            # if not found in either modified or old format consider the file newly added
+            filtered_added_files.add(file_path)
+
+        return filtered_modified_files, filtered_added_files, filtered_old_format
+
+    @staticmethod
+    def specify_files_from_directory(file_set: Set, directory_path: str) -> Set:
+        """Filter a set of file paths to only include ones which are from a specified directory."""
+        filtered_set: Set = set()
+        for file in file_set:
+            if isinstance(file, str) and directory_path in file:
+                filtered_set.add(file)
+
+            # handle renamed files
+            elif isinstance(file, tuple) and directory_path in file[1]:
```

Comment:
so `file` will always be a tuple of length `>= 1` ?

Reply:
If it is a tuple then it represents a renamed file hence it will be in the structure of (old_file_path, new_file_path)